alisa.go

```diff
@@ -65,20 +69,188 @@ func (ali *alisa) createTask(code string) (string, error) {
 	params["ExecTarget"] = ali.envs["ALISA_TASK_EXEC_TARGET"]
 	envBuf, _ := json.Marshal(ali.envs)
 	params["Envs"] = string(envBuf)
-	rsp, err := ali.pop.request(params, ali.popURL, ali.popSecret)
+
+	res, err := ali.requetAndParseResponse(params)
+	if err != nil {
+		return "", -1, err
+	}
+	var val alisaTaskMeta
+	if err = json.Unmarshal(*res, &val); err != nil {
+		return "", -1, err
+	}
+	return val.TaskID, val.Status, nil
+}
+
+// getStatus: returns the task status of taskID
+func (ali *alisa) getStatus(taskID string) (int, error) {
+	params := baseParams(ali.popID)
+	params["Action"] = "GetAlisaTask"
+	params["AlisaTaskId"] = taskID
+
+	res, err := ali.requetAndParseResponse(params)
+	if err != nil {
+		return -1, err
+	}
+	var val alisaTaskStatus
+	if err = json.Unmarshal(*res, &val); err != nil {
+		return -1, err
+	}
+	return val.Status, nil
+}
+
+// completed: check if the status is completed
+func (ali *alisa) completed(status int) bool {
+	return status == 3 || status == 4 || status == 6 || status == 8 || status == 9
+}
+
+// readLogs: reads task logs from `offset`
+// return -1: read to the end
+// return n(>0): keep reading with the offset `n` in the next time
+func (ali *alisa) readLogs(taskID string, offset int) (int, error) {
+	// the `maxLogs` used to deal with too many logs.
+	end := false
+	for i := 0; i < maxLogNum && !end; i++ {
+		params := baseParams(ali.popID)
+		params["Action"] = "GetAlisaTaskLog"
+		params["AlisaTaskId"] = taskID
+		params["Offset"] = fmt.Sprintf("%d", offset)
+
+		res, err := ali.requetAndParseResponse(params)
+		if err != nil {
+			return offset, err
+		}
+		var log alisaTaskLog
+		if err = json.Unmarshal(*res, &log); err != nil {
+			return offset, err
+		}
+		rdLen, err := strconv.Atoi(log.ReadLen)
+		if err != nil {
+			return offset, err
+		}
+		if rdLen == 0 {
+			return offset, nil
+		}
+		offset += rdLen
+		end = log.End
+		fmt.Printf(log.Content)
```

Comment:
Maybe switch `fmt.P*` to `log.Info`, so that the log level can be configured during deployment.